File: Campaign_Samples/go/edit_campaign.go
package main

import (
	b64 "encoding/base64"
	"fmt"
	"io/ioutil"
	"net/http"
	"strings"
)

func main() {
	// Please provide Exotel Sid, APIKey and APIToken from your Exotel account
	exotelSid := "<your Exotel Sid>"
	apiKey := "<your API Key>"
	apiToken := "<your API Token>"

	// Encoding the exotel Sid API_Key , and API Token, used in Authorization header
	encoding := b64.StdEncoding.EncodeToString([]byte(apiKey + ":" + apiToken))
	url := "https://api.exotel.com/v2/accounts/" + exotelSid + "/campaigns/<campaign SID>"

	payload := strings.NewReader(`{ 
		"campaigns": [{ 
			"caller_id": "0XXXXXXXXX1", 
			"url": "http://my.exotel.com/exoml/start_voice/208287", 
			"from": [ "+91XXXXXXXXX4", "+91XXXXXXXX5" ], 
			"status_callback": "http://<callback custom domain>/1gvta9f1", 
			"call_status_callback": "http://<callback custom domain>/1gvta9f1"
			}]
		}`)

	req, _ := http.NewRequest("PUT", url, payload)

	req.Header.Add("Content-Type", "application/json")
	req.Header.Add("Authorization", "Basic "+encoding)

	res, _ := http.DefaultClient.Do(req)

	defer res.Body.Close()
	body, _ := ioutil.ReadAll(res.Body)

	fmt.Println(res)
	fmt.Println(string(body))

}

File: Campaign_Samples/js/edit_campaign.js
var request = require("request");
var accountSid = "<your Exotel Sid>";
var apiKey = "<Your API Key>"
var apiToken = "<Your API Token>";
var encoding = Buffer.from(apiKey + ':' + apiToken).toString('base64')

var options = { method: 'PUT',
  url: 'https://api.exotel.com/v2/accounts/'+ accountSid +'/campaigns/<campaign SID>',
  headers: 
   {
     Authorization: 'Basic ' + encoding,
     'Content-Type': 'application/json' 
    },
  body: 
   { campaigns: 
      [ { caller_id: '0XXXXXXXXX1',
          url: 'http://my.exotel.in/exoml/start/189810',
          from: [ '+91XXXXXXXX4', '+91XXXXXXXXX5' ],
          schedule: { send_at: '2018-11-09T10:00:00+05:30' },
          status_callback: 'http://<callback custom domain>/1gvta9f1',
          call_status_callback: 'http://<callback custom domain>/1gvta9f1' } ] },
  json: true };

request(options, function (error, response, body) {
  if (error) throw new Error(error);

  console.log(body);
  console.log(JSON.stringify(body))
});
